src/dist/replication/meta_server/meta_bulk_load_service.cpp

```diff
@@ -898,6 +974,65 @@ void bulk_load_service::on_partition_ingestion_reply(error_code err,
 }
 
 // ThreadPool: THREAD_POOL_META_STATE
+void bulk_load_service::remove_bulk_load_dir_on_remote_storage(int32_t app_id,
+                                                               const std::string &app_name)
+{
+    std::string bulk_load_path = get_app_bulk_load_path(app_id);
+    _meta_svc->get_meta_storage()->delete_node_recursively(
```

Comment:
maybe you can extract the same code between `remove_bulk_load_dir_on_remote_storage(int32_t app_id, const std::string &app_name)` and `remove_bulk_load_dir_on_remote_storage(std::shared_ptr<app_state> app, bool set_app_not_bulk_loading)`

Reply:
I used to try to combine them into one function, which will make code complex, and it's difficult to extract the similar code for the callback needs different parameters.